### PR TITLE
Fix #1452 Fix #4279 - Remove Attribute from model, and remove interiorVisibleAbsorbtance/exteriorVisibleAbsorbtance

### LINKED ISSUE #1452: Deprecate attributes from Model
Requires replacing all GUI components built on attributes

### LINKED ISSUE #4279: Remove long deprecated methods interiorVisibleAbsorbtance and exteriorVisibleAbsorbtance
Issue overview
--------------

Two of these were supposed to be deprecated in 1.4.0 (release date: July 9, 2014...). They are probably an original typo sor**b**tance v sor**p**tance. Keeping it around hurts the API since the right spelling is so close
 
https://github.com/NREL/OpenStudio/blob/eadff9598068b16479b1f5a223ee0611458d2fc6/src/model/ConstructionBase.hpp#L126-L141

Same in model/PlanarSurface.hpp

This is the commit from 7 years ago that deprecated them (in 1.13.0) https://github.com/NREL/OpenStudio/commit/bd9c9d6103b16789ee8c529ac8d2a6e0b23e3a28

An actual Warning has been issued for 6+ years, so I think it's fair to remove them: https://github.com/NREL/OpenStudio/blob/eadff9598068b16479b1f5a223ee0611458d2fc6/src/model/ConstructionBase.cpp#L304-L307

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

Fix #1452 - Remove Attribute from model
Fix #4279 - Remove interiorVisibleAbsorbtance/exteriorVisibleAbsorbtance from ConstructionBase/PlanarSurface

Please read [OpenStudio Pull Requests](https://github.com/NREL/OpenStudio/wiki/OpenStudio-Pull-Requests) to better understand the OpenStudio Pull Request protocol.

### Pull Request Author

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Checked behavior in OpenStudioApplication, adjusted policies as needed (`src/openstudio_lib/library/OpenStudioPolicy.xml`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/model/CFactorUndergroundWallConstruction.hpp [hunk 1]
```diff
@@ -53,10 +53,11 @@ namespace model {
...
 
-    /** Attribute Name: 'cFactor' */
+    // TODO: why is this stuff virtual? Is this a leftover from the Attribute days?
+
     virtual double cFactor() const;
     virtual bool setCFactor(double cFactor);
-    /** Attribute Name: 'height' */
+
     virtual double height() const;
     virtual bool setHeight(double height);
```
Comment (jmarrec) [inline src/model/CFactorUndergroundWallConstruction.hpp:62]: * [ ] TODO

File: src/model/FFactorGroundFloorConstruction.hpp [hunk 2]
```diff
@@ -53,13 +53,14 @@ namespace model {
...
     virtual double fFactor() const;
     virtual bool setFFactor(double fFactor);
-    /** Attribute Name: 'area' */
+
     virtual double area() const;
     virtual bool setArea(double area);
-    /** Attribute Name: 'perimeterExposed' */
+
     virtual double perimeterExposed() const;
     virtual bool setPerimeterExposed(double perimeterExposed);
```
Comment (jmarrec) [inline src/model/FFactorGroundFloorConstruction.hpp:65]: * [ ] TODO

File: src/model/Facility.hpp [hunk 3]
```diff
@@ -674,15 +564,9 @@ namespace model {
     /// Returns an EndUse object containing all end uses for the simulation.
     boost::optional<EndUses> endUses() const;
 
-    /// Returns the underlying Attribute of the EndUse object.
-    boost::optional<Attribute> endUsesAttribute() const;
-
```
Comment (jmarrec) [inline src/model/Facility.hpp:679]: API Change

File: src/model/Facility.hpp [hunk 4]
```diff
@@ -674,15 +564,9 @@ namespace model {
...
     /// Returns a CalibrationResult object containing data related to model calibration.
     boost::optional<CalibrationResult> calibrationResult() const;
 
-    /// Returns the underlying Attribute of the CalibrationResult object.
-    boost::optional<Attribute> calibrationResultAttribute() const;
-
```
Comment (jmarrec) [inline src/model/Facility.hpp:685]: API Change

File: src/model/PlanarSurface.hpp [hunk 5]
```diff
@@ -170,47 +168,31 @@ namespace model {
...
     boost::optional<double> heatCapacity() const;
 
-    /// Returns interior visible absorptance (unitless), if available.
-    /// Attribute name: interiorVisibleAbsorptance
+    // Returns interior visible absorptance (unitless), if available.
     boost::optional<double> interiorVisibleAbsorptance() const;
 
-    /** \deprecated
-   *   This will be removed after 1.4.0 */
-    boost::optional<double> interiorVisibleAbsorbtance() const;
```
Comment (jmarrec) [inline src/model/PlanarSurface.hpp:201]: API Change

File: src/model/PlanarSurface.hpp [hunk 6]
```diff
@@ -170,47 +168,31 @@ namespace model {
...
-
     /// Returns exterior visible absorptance (unitless), if available.
-    /// Attribute name: exteriorVisibleAbsorptance
     boost::optional<double> exteriorVisibleAbsorptance() const;
 
-    /** \deprecated
-   *   This will be removed after 1.4.0 */
-    boost::optional<double> exteriorVisibleAbsorbtance() const;
```
Comment (jmarrec) [inline src/model/PlanarSurface.hpp:209]: API Change

File: src/model/ConstructionBase.hpp [hunk 7]
```diff
@@ -99,51 +99,35 @@ namespace model {
...
     /** Get the visible absorptance of the interior side of this construction (dimensionless). May
-   *  require simulation ('EnvelopeSummary' table).
-   *
-   *  Attribute Name: 'interiorVisibleAbsorptance' */
+    *  require simulation ('EnvelopeSummary' table). */
     boost::optional<double> interiorVisibleAbsorptance() const;
 
-    /** \deprecated
-   *   This will be removed after 1.4.0 */
-    boost::optional<double> interiorVisibleAbsorbtance() const;
```
Comment (jmarrec) [inline src/model/ConstructionBase.hpp:131]: API Change

File: src/model/ConstructionBase.hpp [hunk 8]
```diff
@@ -99,51 +99,35 @@ namespace model {
...
     /** Get the visible absorptance of the exterior side of this construction (dimensionless). May
-   *  require simulation ('EnvelopeSummary' table).
-   *
-   *  Attribute Name: 'exteriorVisibleAbsorptance' */
+     *  require simulation ('EnvelopeSummary' table). */
     boost::optional<double> exteriorVisibleAbsorptance() const;
 
-    /** \deprecated
-   *   This will be removed after 1.4.0 */
-    boost::optional<double> exteriorVisibleAbsorbtance() const;
```
Comment (jmarrec) [inline src/model/ConstructionBase.hpp:141]: API Change

**Comment by tijcolem** [comment]
Nice job on the cleanup on outdated atributes. looks good to me!

**Review by tijcolem — APPROVED** [review]

**Pull request merged**